src/Stratis.Bitcoin.IntegrationTests/BlockStore/ReOrgRegularlySpecification_Steps.cs

```diff
@@ -0,0 +1,158 @@
+            this.secondNode.FullNode.WalletManager().CreateWallet(WalletPassword, WalletZero);
+            this.thirdNode.FullNode.WalletManager().CreateWallet(WalletPassword, WalletZero);
+            this.fourthNode.FullNode.WalletManager().CreateWallet(WalletPassword, WalletZero);
+
+            this.selfishMiner.CreateRPCClient().AddNode(this.secondNode.Endpoint, true);
+            TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(this.secondNode, this.selfishMiner));
+
+            this.secondNode.CreateRPCClient().AddNode(this.thirdNode.Endpoint, true);
+            TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(this.thirdNode, this.secondNode));
+
+            this.thirdNode.CreateRPCClient().AddNode(this.fourthNode.Endpoint, true);
+            TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(this.fourthNode, this.thirdNode));
+        }
+
+        private void each_mine_a_blocks()
+        {
+            this.sharedSteps.MineBlocks(1, this.selfishMiner, AccountZero, WalletZero, WalletPassword);
+            this.sharedSteps.MineBlocks(1, this.secondNode, AccountZero, WalletZero, WalletPassword);
+            this.sharedSteps.MineBlocks(1, this.thirdNode, AccountZero, WalletZero, WalletPassword);
+            this.sharedSteps.MineBlocks(1, this.fourthNode, AccountZero, WalletZero, WalletPassword);
+        }
+
+        private void selfish_miner_disconnects_and_mines_10_blocks()
+        {
+            this.selfishMiner.FullNode.ConnectionManager.RemoveNodeAddress(this.secondNode.Endpoint);
+            this.selfishMiner.FullNode.ConnectionManager.RemoveNodeAddress(this.thirdNode.Endpoint);
+            this.selfishMiner.FullNode.ConnectionManager.RemoveNodeAddress(this.fourthNode.Endpoint);
+            TestHelper.WaitLoop(() => !TestHelper.IsNodeConnected(this.selfishMiner));
+
+            this.sharedSteps.MineBlocks(10, this.selfishMiner, AccountZero, WalletZero, WalletPassword);
+
+            this.selfishBlockHeight = this.selfishMiner.FullNode.Chain.Height;
+        }
+
+        private void second_node_creates_a_transaction_and_broadcasts()
+        {
+            var thirdNodeReceivingAddress = this.thirdNode.FullNode.WalletManager()
+                .GetUnusedAddress(new WalletAccountReference(WalletZero, AccountZero));
+
+            this.secondNodeTransaction = this.secondNode.FullNode.WalletTransactionHandler().BuildTransaction(
+                SharedSteps.CreateTransactionBuildContext(new WalletAccountReference(WalletZero, AccountZero), WalletPassword, thirdNodeReceivingAddress.ScriptPubKey,
+                    Money.COIN * 1, FeeType.Medium, 101));
+
+            this.secondNode.FullNode.NodeService<WalletController>().SendTransaction(new SendTransactionRequest(this.secondNodeTransaction.ToHex()));
+        }
+
+        private void third_node_mines_this_block()
+        {
+            int feeForSecondToThirdTransaction = 100004520; /////?????
```

Comment:
I can't work out why this amount of satoshis is earned by the miner in addition to the coinbase reward

Reply:
ok, because 1 bitcoin is sent to the address which is mining :)